### PR TITLE
ValidationMessage & Editors: add option to customize VM position

### PR DESCRIPTION
**Pull request by Zedwag**

### CODE REVIEW

File: js/ui/drop_down_editor/ui.drop_down_editor.d.ts [hunk 1]
```diff
@@ -125,6 +126,12 @@ export interface dxDropDownEditorOptions<TComponent> extends dxTextBoxOptions<TC
      * @public
      */
     showDropDownButton?: boolean;
+    /**
+     * @docid
+     * @default null
```
Comment (ksercs) [inline js/ui/drop_down_editor/ui.drop_down_editor.d.ts]: seems like default value should not be null

File: js/ui/validation_message.js [hunk 2]
```diff
@@ -122,30 +122,48 @@ const ValidationMessage = Overlay.inherit({
...
+            default:
+                return [`${positionSide} top`, `${positionSide} bottom`];
+        }
+
+    },
+
     _updatePosition: function() {
         const {
-            positionRequest,
+            positionRequest = 'top',
```
Comment (ksercs) [inline js/ui/validation_message.js]: why do we need this initial value here?

File: js/ui/validation_message.js [hunk 3]
```diff
@@ -122,30 +122,48 @@ const ValidationMessage = Overlay.inherit({
         this.option({ maxWidth });
     },
 
+    _getPositionsArray: function(positionRequest, positionSide) {
```
Comment (ksercs) [inline js/ui/validation_message.js]: let's rename `positionRequest` to `position` everywhere

File: testing/testcafe/tests/editors/checkBox/validationMessage.ts [hunk 4]
```diff
@@ -40,3 +42,40 @@ test('ValidationMessage integrated in editor should not raise any errors when it
...
+    test(`CheckBox ValidationMessage position is correct (${position}, ${theme})`, async (t) => {
+      const checkBox1 = new CheckBox('#container');
+      await t
+        .click(checkBox1.element)
+        .click(checkBox1.element)
+        .expect(true).ok();
+
+      await t.expect(await compareScreenshot(t, `checkbox-validation-message-position=${position},theme=${theme.replace(/\./g, '-')}.png`)).ok();
+    }).before(async () => {
+      await changeTheme(theme);
```
Comment (ksercs) [inline testing/testcafe/tests/editors/checkBox/validationMessage.ts:67]: let's resize window size here to make screenshot smaller

File: testing/testcafe/tests/editors/pages/validationMessagePosition.html [hunk 5]
```diff
@@ -0,0 +1,83 @@
+<!DOCTYPE html>
```
Comment (ksercs) [inline testing/testcafe/tests/editors/pages/validationMessagePosition.html]: seems like u don't use this page anywhere

File: js/ui/editor/editor.d.ts [hunk 6]
```diff
@@ -62,6 +63,12 @@ export interface EditorOptions<TComponent> extends WidgetOptions<TComponent> {
      * @public
      */
     validationMessageMode?: ValidationMessageMode;
+    /**
+     * @docid
+     * @default "bottom"
+     * @public
+     */
+     validationMessagePosition?: ValidationMessagePosition;
```
Comment (ksercs) [inline js/ui/editor/editor.d.ts]: "auto" value type should be published only in components where it's useful. It it you useful for slider or CheckBox?

File: testing/testcafe/tests/editors/pages/validationMessagePosition.html (hunk 5 above)
Comment (Zedwag) [reply testing/testcafe/tests/editors/pages/validationMessagePosition.html]: yeah, forgot this one

File: testing/testcafe/tests/editors/checkBox/validationMessage.ts (hunk 4 above)
Comment (Zedwag) [reply testing/testcafe/tests/editors/checkBox/validationMessage.ts:67]: Here we go again

File: js/ui/drop_down_editor/ui.drop_down_editor.d.ts (hunk 1 above)
Comment (Zedwag) [reply js/ui/drop_down_editor/ui.drop_down_editor.d.ts]: yep

File: js/ui/validation_message.js (hunk 2 above)
Comment (Zedwag) [reply js/ui/validation_message.js]: made default option `top`

File: js/ui/validation_message.js (hunk 3 above)
Comment (Zedwag) [reply js/ui/validation_message.js]: mmkey

File: js/ui/editor/editor.d.ts (hunk 6 above)
Comment (Zedwag) [reply js/ui/editor/editor.d.ts]: done

File: testing/testcafe/tests/editors/checkBox/validationMessage.ts (hunk 4 above)
Comment (Zedwag) [reply testing/testcafe/tests/editors/checkBox/validationMessage.ts:67]: done

File: js/ui/validation_message.js (hunk 3 above)
Comment (Zedwag) [reply js/ui/validation_message.js]: renamed to `positionSide` becase VM already has `position` option

File: js/ui/drop_down_editor/ui.drop_down_editor.d.ts [hunk 7]
```diff
@@ -125,6 +126,12 @@ export interface dxDropDownEditorOptions<TComponent> extends dxTextBoxOptions<TC
      * @public
      */
     showDropDownButton?: boolean;
+    /**
+     * @docid
+     * @default 'auto'
```
Comment (IlyaKhD) [inline js/ui/drop_down_editor/ui.drop_down_editor.d.ts:132]: Is this still relevant? `'auto'` is not declared here:
```typescript
  validationMessagePosition?: ValidationMessagePosition;
```

File: js/common.d.ts [hunk 8]
```diff
@@ -256,6 +256,12 @@ export type TooltipShowMode = 'always' | 'onHover';
  */
 export type ValidationMessageMode = 'always' | 'auto';
 
+/**
+ * @public
+ * @namespace DevExpress.common
+ */
+ export type ValidationMessagePosition = 'top' | 'right' | 'bottom' | 'left';
+
```
Comment (IlyaKhD) [inline js/common.d.ts]:
```suggestion
```
Use `Position` instead

File: js/ui/editor/editor.d.ts [hunk 9]
```diff
@@ -62,6 +63,12 @@ export interface EditorOptions<TComponent> extends WidgetOptions<TComponent> {
      * @public
      */
     validationMessageMode?: ValidationMessageMode;
+    /**
+     * @docid
+     * @default "bottom"
+     * @public
+     */
+     validationMessagePosition?: Position;
```
Comment (IlyaKhD) [inline js/ui/editor/editor.d.ts]: Remove extra space
```suggestion
    validationMessagePosition?: Position;
```

Comment (Zedwag) [reply js/ui/editor/editor.d.ts]: done

File: js/ui/drop_down_editor/ui.drop_down_editor.d.ts [hunk 10]
```diff
@@ -125,6 +126,12 @@ export interface dxDropDownEditorOptions<TComponent> extends dxTextBoxOptions<TC
...
+     * @public
+     */
+    validationMessagePosition?: Position | 'auto';
```
Comment (IlyaKhD) [inline js/ui/drop_down_editor/ui.drop_down_editor.d.ts]:
```diff
import {
    ApplyValueMode,
+   Mode,
    Position,
} from '../../common';
```

```suggestion
    validationMessagePosition?: Position | Mode;
```

Comment (Zedwag) [reply js/ui/drop_down_editor/ui.drop_down_editor.d.ts]: done

**Review by ksercs — APPROVED** [review]

**Review by EugeniyKiyashko — APPROVED** [review]

**Pull request merged**